utils/utils.go

```diff
@@ -102,3 +103,49 @@ func CheckValidQuery(transferUrl *url.URL) (err error) {
 
 	return errors.New("invalid query parameter(s) " + transferUrl.RawQuery + " provided in url " + transferUrl.String())
 }
+
+// Applies a \24 bit mask to an IPv4 address
+// If the input string isn't an IPv4 address then the input string is returned
+func IPv4Mask(ipStr string) string {
+	ip := net.ParseIP(ipStr).To4()
+	if ip == nil {
+		return ipStr
+	}
+	mask := net.CIDRMask(24, 32)
+	maskedIP := ip.Mask(mask)
+	return maskedIP.String()
+}
+
+// Applies a \64 bit mask to an IPv4 address
+// If the input string isn't an IPv4 address then the input string is returned
+func IPv6Mask(ipStr string) string {
+	ip := net.ParseIP(ipStr).To16()
+	if ip == nil || ip.To4() != nil {
+		return ipStr
+	}
+	mask := net.CIDRMask(64, 128)
+	maskedIP := ip.Mask(mask)
+	return maskedIP.String()
+}
+
+// MaskIP applies the appropriate subnet mask to the input IP address.
+func MaskIP(ipStr string) string {
+	ip := net.ParseIP(ipStr)
+	if ip == nil {
+		// Invalid IP
```

Comment:
Does this cause an explosion in cardinality if a serves is using hostnames and not an IP?

Good habits would have an indication to the caller that a special case occurred. Maybe also return a Boolean?